### PR TITLE
Dịch chương 25

### PR DESCRIPTION
**Pull request by naml3i**
## Nếu bạn tạo Pull Request để dịch, vui lòng hoàn thành các thông tin dưới đây.

* [x] Đánh dấu [x] (không có dấu cách quanh x) thể hiện dấu tick vào các câu dưới đây cho các câu trả lời 'Có' hoặc 'Đúng'

* [x] Pull Request này được link tới một issue? Đánh số issue đó *ngay sau* dấu #189 

* [x] Bạn có kiến thức về Machine Learning không? Chúng tôi ưu tiên những người có trước kiến thức về Machine Learning để việc review được thuận tiện hơn.

* [x] Bạn đã đọc [Bảng thuật ngữ](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/glossary.md) chưa? Bạn cần dịch các thuật ngữ theo bảng thuật ngữ chung.

* [x] Tất cả các thuật ngữ trong chương này đã có trong [Bảng thuật ngữ](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/glossary.md) chưa? Nếu chưa, bạn có thể tạo một issue hoặc pull request mới để bổ sung nghĩa của từ đó vào Bảng thuật ngữ.

* [x] Pull request này đã sẵn sàng cho việc review chưa? Nếu đã sẵn sàng, bạn vui lòng gán label 'review wanted' cho pull request này. Nếu chưa, bạn vui lòng thêm cụm "[WIP]" (work in progress) vào trước tiêu đề của pull request.

## Lưu ý 
**Chỉ bắt đầu review khi mọi mục trên đây đều đã được tick.**

### CODE REVIEW

**Comment by naml3i** [comment]
- Chữ regularization được giữ nguyên vì hiện thời trong glossary nó được giữ nguyên. (Mình thấy dịch là chuẩn hóa cũng ổn mà?). Chương này nó được dùng hơi nhiều nên đọc lên liên tiếp nghe hơi bị rổn rảng. 
- Chữ dropout diễn giải tường minh ra là một khái niệm dài, mình không dịch được, mong mọi người góp ý, hoặc là để nguyên. #189

**Comment by ngcthuong** [comment]
> * Chữ regularization được giữ nguyên vì hiện thời trong glossary nó được giữ nguyên. (Mình thấy dịch là chuẩn hóa cũng ổn mà?). Chương này nó được dùng hơi nhiều nên đọc lên liên tiếp nghe hơi bị rổn rảng.
> * Chữ dropout diễn giải tường minh ra là một khái niệm dài, mình không dịch được, mong mọi người góp ý, hoặc là để nguyên. #189

Regularization hiểu theo nghĩa là giới hạn không gian nghiệm lại theo một điều kiện nào đó. Còn "chuẩn hoá" là standardization nghĩa hoàn toàn khác nhau.

File: chapters/ch25.md [hunk 1]
```diff
@@ -1,31 +1,32 @@
 # Techniques for reducing avoidable bias
 ->
-
+# Các kĩ thuật để giảm độ chệch tránh được
 
 If your learning algorithm suffers from high avoidable bias, you might try the following techniques:
 ->
-
+Nếu thuật toán học của bạn gặp vấn đề với độ chệch tránh khỏi lớn, bạn có thể thử những kĩ thuật sau:
```
Comment (william-vu) [inline chapters/ch25.md]: > độ chệch tránh khỏi

ở tiêu đề là "độ chệch tránh được"

Thêm nữa; theo mình thì nên dịch là "độ chệch có thể tránh được", vì cái này chỉ là "có thể" tránh được chứ chưa chắc đã "tránh được". Cụ thể hơn, trong ngữ cảnh này là nếu như model được huấn luyện đủ tốt thì có cơ hội training error rate sẽ bằng optimal error rate & nhưng có thể mất rất nhiều time để tìm được model tốt & cách huấn luyện tốt. (ví dụ của model: CNN; ví dụ của cách huấn luyện: sgd, ada)

File: chapters/ch25.md [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
 
 * **Increase the model size** ​(such as number of neurons/layers): This technique reduces bias, since it should allow you to fit the training set better. If you find that this increases variance, then use regularization, which will usually eliminate the increase in variance.
 ->
+* **Tăng kích thước mẫu** (ví dụ như số lượng neuron/tầng): Kĩ thuật này làm giảm độ chệch, vì nó cho phép fit tập huấn luyện tốt hơn. Nếu bạn thấy việc này làm tăng phương sai, hãy sử dụng regularization, vốn thường triệt tiêu việc tăng phương sai.
```
Comment (william-vu) [inline chapters/ch25.md]: "model size" theo mình hiểu không phải là **kích thước mẫu**. Ví dụ đơn giản của tăng model size là mạng MLP bạn tăng số layer lên, qua đó model có nhiểu parameters hơn và sẽ fit được tập huấn luyện (lớn?) tốt hơn.

File: chapters/ch25.md [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
 ->
-
+* **Thay đổi các đặc trưng đầu vào dựa trên nhận thức rõ có được từ phân tích lỗi ​**: Giả sử việc phân tích lỗi gợi ý bạn rằng hãy tạo thêm những đặc trung bổ sung nhằm giúp thuật toán có thể loại bỏ một kiểu lỗi đặc thù. (Chúng ta sẽ bàn vấn đề này kĩ hơn ở chương sau.) Những đặc trưng mới này có thể hiệu quả với cả độ chệch và phương sai. Trong lý thuyết, thêm đặc trưng có thể làm tăng phương sai; nhưng nếu bạn thấy đúng là điều đó xảy ra, hãy sử dụng regularization, vốn thường triệt tiêu việc tăng phương sai.
 
 * **Reduce or eliminate regularization​** (L2 regularization, L1 regularization, dropout): This will reduce avoidable bias, but increase variance.
 ->
-
+* **Giảm hoặc triệt tiêu regularization​** (L2 regularization, L1 regularization, dropout): Việc này sẽ làm giảm độ chệch có thể tránh được, nhưng sẽ làm tăng phương sai.
```
Comment (william-vu) [inline chapters/ch25.md]: dùng "triệt tiêu" chắc cũng ok. Nhưng với mình khi nhìn thấy từ triệt tiêu thì mình nghĩ đến công thức kiểu như a-b, hai cái đó triệt tiêu nhau; trong ngữ cảnh này thì "triệt tiêu" = "loại bỏ"?

File: chapters/ch25.md [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
 
 * **Modify model architecture​** (such as neural network architecture) so that it is more suitable for your problem: This technique can affect both bias and variance.
 ->
-
+* **Thay đổi kiến trúc mẫu** (ví dụ như kiến trúc mạng neural) để nó trở nên phù hợp hơn với bài toán của bạn: Kĩ thuật này có thể ảnh hưởng đến cả độ chệch và phương sai.
```
Comment (william-vu) [inline chapters/ch25.md]: > **Thay đổi kiến trúc mẫu** 

same comment giống như "tăng kích thước mẫu" ở trên.

**Review by william-vu — COMMENTED** [review]
Mình có để lại một số comments sau khi đọc bản dịch này.  Hi vọng nó có ích.

**Comment by naml3i** [comment]
@ngcthuong chữ "chuẩn hóa" ở đây không phải là "tiêu chuẩn hóa" như standardization. Khi mình nói chuẩn hóa ý niệm của mình nó có nét tương tự như "normalization" vậy. 

Có điểm khác biệt là normalization sẽ scale lại data input đầu vào, VD chuyển hết range của các features về [0, 1], còn regularization là một kĩ thuật để tránh overfitting, thay vì scaling lại data thì nó điều chỉnh hàm predict. Thực ra theo một nghĩa nào đó, việc này cũng là "chặn trên chặn dưới", áp đặt một cái range cho các giá trị của hệ số của hàm predict, tức là nó cũng "chuẩn hóa" hệ số vào trong một dải.

(Đây chỉ là trao đổi chút về mặt chữ/ngữ nghĩa, còn đã thống nhất trong glossary thì sẽ để nguyên. Dùng chung là chuẩn hóa thì sẽ lẫn lộn, mà chưa thể có được term nào thay thế thì để nguyên thôi).

File: chapters/ch25.md (hunk 2 above)
Comment (naml3i) [reply chapters/ch25.md]: À chỗ này mình sơ suất. Nó phải là "tăng kích thước mô hình" :-) 👍

File: chapters/ch25.md (hunk 3 above)
Comment (naml3i) [reply chapters/ch25.md]: Chữ "loại bỏ" tốt hơn nhiều. 👍 
Khi mình dịch mình cũng đã lăn tăn chữ "triệt tiêu" vì đúng là nó mang sắc thái trong toán có vẻ không tự nhiên lắm ở đây (VD phân số có cùng thừa số a trên tử và mẫu nên a "triệt tiêu") -- mà không nghĩ ra alternative choice đơn giản thế này.

File: chapters/ch25.md (hunk 4 above)
Comment (naml3i) [reply chapters/ch25.md]: Đã sửa tương ứng 👍

File: chapters/ch25.md (hunk 1 above)
Comment (tiepvupsu) [reply chapters/ch25.md]: @naml3i bạn vẫn chưa "resolve" comment này nhưng đã "marked as resolved".

Mình thấy comment của @william-vu khá hợp lý. Nếu bạn không đồng ý thì nên phản biện lại.

File: chapters/ch25.md [hunk 5]
```diff
@@ -1,31 +1,32 @@
...
 
 One method that is not helpful:
 ->
-
+Một phương pháp không hiệu quả:
 
 * **Add more training data​**: This technique helps with variance problems, but it usually has no significant effect on bias.
 ->
-
+* **Thêm dữ liệu huấn luyện​**: Kĩ thuật này có ích với vấn đề  phương sai, nhưng nó thường không có tác dụng gì đáng kể với độ chệch.
```
Comment (tiepvupsu) [inline chapters/ch25.md]: hai dấu cách "đề  phương"

**Review by tiepvupsu — CHANGES_REQUESTED** [review]

File: chapters/ch25.md (hunk 1 above)
Comment (william-vu) [reply chapters/ch25.md]: hình như bạn ấy resolve rồi mà bác. Em có xem bản mới nhất bạn ấy up lên. Bác double check hộ em.

Comment (william-vu) [reply chapters/ch25.md]: e vừa check lại, bạn ấy mới resolve là tiêu đề, chưa sửa cái ở bên dưới.

Comment (sonvx) [reply chapters/ch25.md]: -> a unresolved lại rồi nhé.

File: chapters/ch25.md [hunk 6]
```diff
@@ -1,31 +1,32 @@
...
 
 
 * **Modify input features based on insights from error analysis​**: Say your error analysis inspires you to create additional features that help the algorithm eliminate a particular category of errors. (We discuss this further in the next chapter.) These new features could help with both bias and variance. In theory, adding more features could increase the variance; but if you find this to be the case, then use regularization, which will usually eliminate the increase in variance.
 ->
-
+* **Thay đổi các đặc trưng đầu vào dựa trên nhận thức rõ có được từ phân tích lỗi ​**: Giả sử việc phân tích lỗi gợi ý bạn rằng hãy tạo thêm những đặc trưng bổ sung nhằm giúp thuật toán có thể loại bỏ một kiểu lỗi đặc thù. (Chúng ta sẽ bàn vấn đề này kĩ hơn ở chương sau.) Những đặc trưng mới này có thể hiệu quả với cả độ chệch và phương sai. Trong lý thuyết, thêm đặc trưng có thể làm tăng phương sai; nhưng nếu bạn thấy đúng là điều đó xảy ra, hãy sử dụng regularization, vốn thường loại bỏ việc tăng phương sai.
```
Comment (sonvx) [inline chapters/ch25.md]: '**eliminate a particular category of errors.**' bạn đang dịch là '**loại bỏ một kiểu lỗi đặc thù**' thì chưa chuẩn lắm. Nên dịch là '**loại bỏ một lớp/nhóm các kiểu lỗi đặc thù**' thì sát nghĩa hơn.

File: chapters/ch25.md (hunk 5 above)
Comment (ngcthuong) [reply chapters/ch25.md]: > @ngcthuong chữ "chuẩn hóa" ở đây không phải là "tiêu chuẩn hóa" như standardization. Khi mình nói chuẩn hóa ý niệm của mình nó có nét tương tự như "normalization" vậy.
> 
> Có điểm khác biệt là normalization sẽ scale lại data input đầu vào, VD chuyển hết range của các features về [0, 1], còn regularization là một kĩ thuật để tránh overfitting, thay vì scaling lại data thì nó điều chỉnh hàm predict. Thực ra theo một nghĩa nào đó, việc này cũng là "chặn trên chặn dưới", áp đặt một cái range cho các giá trị của hệ số của hàm predict, tức là nó cũng "chuẩn hóa" hệ số vào trong một dải.
> 
> (Đây chỉ là trao đổi chút về mặt chữ/ngữ nghĩa, còn đã thống nhất trong glossary thì sẽ để nguyên. Dùng chung là chuẩn hóa thì sẽ lẫn lộn, mà chưa thể có được term nào thay thế thì để nguyên thôi).

Để mình giải thích thêm một chút. Thuật ngữ regularization dùng trong optimization từ trước deep learning hay ML. Và nghĩa khái quát rộng hơn - như mình nói- là giới hạn không gian nghiệm theo một điều kiện ràng buộc nào đó. Khác biệt nhiều với nghĩa chuẩn hoá hay là "normalizartion"

Vì trong các thuật ngữ khác như "total variation regularization" mình dịch regularization thảnh "chuẩn hoá" thì bị lệch với nghĩa gốc. Nên mình nghĩ không nên dùng tử "chuẩn hoá" là lý do như vậy.

Comment (ngcthuong) [reply chapters/ch25.md]: Và từ revgularization hay được dùng theo nghĩa tương ứng với "constraint"

Comment (tiepvupsu) [reply chapters/ch25.md]: @ngcthuong nó có "tác dụng" giống "constraint" còn nghĩa thì không hẳn giống.
"Chuẩn hóa" thì chắc chắc không đúng rồi.

Tạm thời bỏ qua các dịch "regularization" ở chương này, giữ nguyên là "regularization" như từ đầu. Mình có thể tạo hẳn một PR khác để bàn riêng về từ này, khi chấp nhận thì tạo một commit để thay toàn bộ các từ từ đầu.

**Comment by tiepvupsu** [comment]
@naml3i bạn có tiếp tục dịch chương này được không? Milestone của chương này là thứ ba tuần sau, hy vọng PR này được merge trước ngày đó.

**Comment by naml3i** [comment]
@tiepvupsu 2 ngày rồi là day off của mình ;) Mình đã check lại và update.

File: chapters/ch25.md (hunk 6 above)
Comment (naml3i) [reply chapters/ch25.md]: Agreed, "một nhóm/lớp" có vẻ bám sát hơn với "**category**" hơn là "kiểu lỗi" (~ **type** of error), nhưng "nhóm các kiểu" có vẻ hơi thừa? --> Hiện thời chuyển thành **"một nhóm các lỗi đặc thù"**.

File: chapters/ch25.md (hunk 1 above)
Comment (naml3i) [reply chapters/ch25.md]: - Đã sửa. (Thực ra mình đã sửa hết theo suggestion của @william-vu mà commit nhầm, khi làm lại lại sót :D).

- Có một điểm mình chưa thấy ổn lắm là cả cụm "high avoidable bias", đọc tiếng Anh thì không sao (vì avoidable là một từ có mức compound rất lớn), khi sang tiếng Việt nó bị dàn ra quá lại kèm với từ "lớn" sau đó: "độ chệch có thể tránh được lớn", nên mình đề xuất để nghiêng cụm "avoidable bias" để có thể nhận định rõ "độ chệch có thể tránh được" là một khái niệm, "lớn" là tính từ bổ sung: "nếu thuật toán học của bạn gặp vấn đề với _độ chệch có thể tránh khỏi_ lớn (...)"

(Cái italic này không có trong bản gốc, nhưng cũng là một kĩ thuật thường gặp để 1 cụm dài nó stands out và đi với nhau nguyên vẹn, thay vì dùng ngoặc kép hay gạch nối. Mọi người cho ý kiến. Nếu ở các chương trước người đọc đã gặp nhiều khái niệm này thì chắc không cần phải in nghiêng).

**Review by aivivn — APPROVED** [review]
Cảm ơn @naml3i rất nhiều. Chương này đã đảm bảo yêu cầu của giai đoạn một.

**Review by tiepvupsu — APPROVED** [review]
